docs/spark-queries.md

```diff
@@ -126,6 +126,8 @@ To select a specific table snapshot or the snapshot at some time in the DataFram
 
 * `snapshot-id` selects a specific table snapshot
 * `as-of-timestamp` selects the current snapshot at a timestamp, in milliseconds
+* `branch` selects the head snapshot of the specified branch. Note that currently branch cannot be combined with as-of-timestamp.
+* `tag` selects the snapshot associated with the specified tag
```

Comment:
do we need to mention that tag also cannot be combined with as-of-timestamp.

Reply:
or we can wait till https://github.com/apache/iceberg/pull/6575 gets merged. So that we don't have to mention it for both branch and tag. But we need to add an example in ##SQL also.

Reply:
Definitely agree on having a SQL example once #6575 gets merged. For combining as-of-timestamp with tag I felt that was apparent since a tag can only map to a single snapshot which conflicts with passing in a timestamp, where as branch + time travel is a different case.

Reply:
Given that is a syntax change, I am waiting for more time for others to take a look. I think we can first merge this one and add that later.